indent/javascript.vim

```diff
@@ -240,7 +240,7 @@ function s:LineHasOpeningBrackets(lnum)
     endif
     let pos = match(line, '[][(){}]', pos + 1)
   endwhile
-  return (open_0 > 0) . (open_2 > 0) . (open_4 > 0)
+  return (open_0 > 0?1:(open_0 == 0 ? 0 : 2)) . (open_2 > 0) . (open_4 > 0)
```

Comment:
This function returned a string of three 1 or 0 digits. if there was more opening than closing '({[' brackets one or more of these numbers would be one and if not the resulting char would be 0,this just allows distinction if there were instead more closing ')' on a line by resulting in a 2 as the first character of the string(curly is the second character, square brackets the third)